app/src/main/java/com/codingblocks/cbonlineapp/services/DownloadService.kt

```diff
@@ -57,15 +58,13 @@ class DownloadService : IntentService("Download Service"), AnkoLogger,
             .setOngoing(true) // THIS is the important line
             .setAutoCancel(false)
     }
-    private lateinit var database: AppDatabase
     private lateinit var contentDao: ContentDao
 
     override fun onHandleIntent(intent: Intent) {
         val title = intent.getStringExtra("title")
         dataId = intent.getStringExtra("id")
         notificationBuilder.setContentTitle(title)
-        database = AppDatabase.getInstance(this)
-        contentDao = database.contentDao()
+        contentDao = AppDatabase.getInstance(this).contentDao()
```

Comment:
there should be not dependacy of AppDatabase anywhere
Infact i told you to delete the singleton object and use koin everywhere